h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -128,17 +128,22 @@
     _persist = 0;               // No persistance
   }
 
-  /** Incoming TCP request.  Make a read-mode AutoBuffer from the open Channel,
-   *  figure the originating H2ONode from the first few bytes read. */
-  AutoBuffer( SocketChannel sock ) throws IOException {
+  /** Incoming TCP request.  Make a read-mode AutoBuffer from the open Channel.
+   * If isSenderH2ONode is set to true than also figure the originating H2ONode from the first few bytes read, otherwise
+   * just don't set _h2o field because the communication originates from non-h2o node such as spark executor*/
+  AutoBuffer( SocketChannel sock, boolean isSenderH2ONode ) throws IOException {
     _chan = sock;
     raisePriority();            // Make TCP priority high
     _bb = BBP_BIG.make();       // Get a big / TPC-sized ByteBuffer
     _bb.flip();
     _read = true;               // Reading by default
     _firstPage = true;
     // Read Inet from socket, port from the stream, figure out H2ONode
-    _h2o = H2ONode.intern(sock.socket().getInetAddress(), getPort());
+    if(isSenderH2ONode) {
```

Comment:
@mmalohlava after merging with my branch it should be OK on the H2O core side since I don't see anywhere where @MadMan0708 would create new sockets so he will be using proper (non)ssl ones.

The problem, which I need to check, might start on the SW side, I need to check again how he establishes communication there - sockets there have to be wrapped with the same `SSLEngine`.